lib/ansible/galaxy/collection/__init__.py

```diff
@@ -1894,10 +1923,10 @@ def _resolve_depenency_map(
         error_msg_lines = [
             (
                 'Failed to resolve the requested dependencies map. '
-                'Got the candidate {req.fqcn!s}:{req.ver!s} ({dep_origin!s}) '
-                'which didn\'t satisfy all of the following requirements:'.
+                'A package index offered us an invalid candidate '
+                f'{dep_exc.candidate !s} ({{dep_origin!s}}) and it does '
+                'not satisfy all of the following requirements:'.
```

Comment:
Why change this? I think the original is fine.

Reply:
The original is misleading. It made me think that the problem is always caused by invalid metadata. However, this is actually about a bug in the package index and the fact that it is probably lying to us. I didn't rully understand this until I started digging into when this exception is actually raised. So I wanted to make it clear what's happening.

Reply:
I don't understand how the original is misleading or how this clarifies it.